### PR TITLE
EVAKA-3968: mobile attendances v2

### PR DESCRIPTION
**Pull request by Joosakur**
#### Checklist for pull request creator

- [ ] The code is consistent with the existing code base
- [ ] Tests have been written for the change (e2e, integration tests)
- [ ] The change has been tested locally
- [ ] The change conforms to the UX specifications
- [ ] The code is self-documenting or has been documented sufficiently, e.g., in the README
- [ ] The branch has been rebased against master before the PR was created

#### Checklist for pull request reviewer (copy to review text box)

```
- [ ] All changes in all changed files have been reviewed
- [ ] The code is consistent with the existing code base
- [ ] Tests have been written for the change (e2e, integration tests)
- [ ] The change has been tested locally
- [ ] The change conforms to the UX specifications
- [ ] The code is self-documenting or has been documented sufficiently, e.g., in the README
- [ ] The PR branch has been rebased against master and force pushed if necessary before merging
```

### CODE REVIEW

File: frontend/packages/employee-frontend/src/api/attendances.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      return {
+        unit: data.unit,
+        children: data.children.map((attendanceChild) => {
+          return {
+            ...attendanceChild,
+            attendance: attendanceChild.attendance
+              ? {
+                  ...attendanceChild.attendance,
+                  arrived: new Date(attendanceChild.attendance.arrived),
+                  departed: new Date(attendanceChild.attendance.departed)
```
Comment (Joosakur) [inline frontend/packages/employee-frontend/src/api/attendances.ts:74]: does this need null check? `departed: attendanceChild.attendance.departed ? new Date(attendanceChild.attendance.departed) : null`

File: frontend/packages/employee-frontend/src/api/attendances.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+export async function returnToComing(
+  unitId: UUID,
+  childId: UUID
+): Promise<void> {
+  return client
+    .post<void>(
+      `/attendances/units/${unitId}/children/${childId}/return-to-coming`
+    )
+    .then((res) => res.data)
```
Comment (Joosakur) [inline frontend/packages/employee-frontend/src/api/attendances.ts:134]: signature says `Promise<void>` but this still returns `res.data` --> remove unnecessary` .then` and replace `return` with `await`.

File: frontend/packages/employee-frontend/src/components/attendances/Absences.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+interface ChildListItemProps {
+  attendanceChild: AttendanceChild
+}
+
+export default React.memo(function Absences({
+  attendanceChild
+}: ChildListItemProps) {
+  const { i18n } = useTranslation()
+
```
Comment (Joosakur) [inline frontend/packages/employee-frontend/src/components/attendances/Absences.tsx:48]: I think there's room for simplification here and this does not yet support clubs, but ok for now

File: frontend/packages/employee-frontend/src/components/attendances/ChildListItem.tsx [hunk 4]
```diff
@@ -92,32 +92,32 @@ export default React.memo(function ChildListItem({
...
-              childInGroup.arrived &&
+            {attendanceChild.status === 'PRESENT' &&
+              attendanceChild.attendance?.arrived &&
               `${
-                childInGroup.arrived?.getHours() < 10
-                  ? `0${childInGroup.arrived?.getHours()}`
-                  : childInGroup.arrived?.getHours()
+                attendanceChild.attendance.arrived.getHours() < 10
+                  ? `0${attendanceChild.attendance.arrived.getHours()}`
+                  : attendanceChild.attendance.arrived?.getHours()
```
Comment (Joosakur) [inline frontend/packages/employee-frontend/src/components/attendances/ChildListItem.tsx:105]: probably easier with some format function

File: frontend/packages/employee-frontend/src/state/attendance-ui.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      groupIdOrAll: UUID | 'all'
+    ) {
+      if (isSuccess(attendanceResponse)) {
+        if (groupIdOrAll !== 'all')
+          attendanceResponse.data.children = attendanceResponse.data.children.filter(
+            (child) => child.groupId === groupIdOrAll
+          )
+      }
+      setAttendanceResponse(attendanceResponse)
+    }
```
Comment (Joosakur) [inline frontend/packages/employee-frontend/src/state/attendance-ui.tsx:49]: I think we should not filter in setter, but rather offer a selector, i.e. a filtered getter

File: frontend/packages/employee-frontend/src/components/attendances/Absences.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        )
+      : undefined
+
+  const daycareAbsences =
+    attendanceChild.placementType === 'DAYCARE'
+      ? attendanceChild.absences.filter(
+          (absence) => absence.careType === 'DAYCARE'
+        )
+      : undefined
+
```
Comment (Joosakur) [inline frontend/packages/employee-frontend/src/components/attendances/Absences.tsx:69]: in addition to club theres other placement types not considered (2x preparatory and part time daycare)

File: frontend/packages/employee-frontend/src/state/attendance-ui.tsx (hunk 5 above)
Comment (massproduction) [reply frontend/packages/employee-frontend/src/state/attendance-ui.tsx:49]: Added todo to fix this

**Review by massproduction — APPROVED** [review]

**Pull request merged**